assignment3/GridWordSearcher.java

```diff
@@ -0,0 +1,121 @@
+package assignment3;
+
+import java.util.*;
+
+public class GridWordSearcher {
+    private int n;
+    private int m;
+    private char[][] grid;
+    private WordDictionary dict;
+    private static List<List<Integer>> directions;
+
+    static {
+        directions = new ArrayList<>();
+        for (int i = -1; i <= 1; i++) {
+            for (int j = -1; j <= 1; j++) {
+                if (i == 0 && j == 0) {
+                    continue;
+                }
+                directions.add(Arrays.asList(i, j));
+            }
+        }
+    }
+
+    private static class GridPosition {
+        private int i;

```

Comment:
It would be good to give these meaningful names instead of i and j. Maybe row and column?